File: app/src/main/java/com/mss/imagesearcher/di/DBModule.java
package com.mss.imagesearcher.di;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.room.Room;

import com.mss.imagesearcher.domain.repositories.CacheDBRepository;
import com.mss.imagesearcher.repositories.db.CacheDB;
import com.mss.imagesearcher.repositories.db.CacheDBRepositoryImpl;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class DBModule {

    private static final String DATABASE_NAME = "CACHE";

    @Singleton
    @Provides
    @NonNull
    CacheDB provideCacheDB(@NonNull final Context context) {
        return Room.databaseBuilder(context, CacheDB.class, DATABASE_NAME).build();
    }

    @Singleton
    @Provides
    @NonNull
    CacheDBRepository provideCacheDBRepository(@NonNull final CacheDB cacheDB) {
        return new CacheDBRepositoryImpl(cacheDB);
    }
}

File: app/src/main/java/com/mss/imagesearcher/di/DBModule.kt
package com.mss.imagesearcher.di

import android.content.Context
import androidx.room.Room
import com.mss.imagesearcher.model.repositories.DBRepository
import com.mss.imagesearcher.model.repositories.db.AppDatabase
import com.mss.imagesearcher.model.repositories.db.DBRepositoryImpl
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class DBModule {

    @Singleton
    @Provides
    fun provideDatabase(context: Context): AppDatabase {
        return Room.databaseBuilder(context, AppDatabase::class.java, "history")
                .fallbackToDestructiveMigration()
                .build()
    }

    @Singleton
    @Provides
    fun provideDBRepository(appDatabase: AppDatabase): DBRepository {
        return DBRepositoryImpl(appDatabase)
    }
}